### PR TITLE
Initial test attachments functionality

### PR DESCRIPTION
**Pull request by wentwrong**
## Purpose
By default, Jenkins has no option to show test attachments (screenshots and videos in case of testcafe). Currently [JUnit attachments plugin](https://github.com/jenkinsci/junit-attachments-plugin) is being used for this functionality, but specifically for Java test-runners in mind. Unmaintainability, along with several functions not working correctly (i.e. broken links, inability to display attachments with the same filenames) make plugin's usage undesirable.

## Features
- Running from pipeline as part of JUnit plugin (custom publisher). E.g.:
```
pipeline {
  agent any
  stages {
    stage('Test') {
      steps {
        sh 'testcafe firefox e2e/**/* -r jenkins:res.xml -S --video video-records'
      }
    }
  }
  post {
    always {
      junit testDataPublishers: [[$class: 'TestcafePublisher']], 
            testResults: 'res.xml'
    }
  }
}
```
- Running from GUI as additional report feature in JUnit test result publishing:

AdditionalTestReportFeatures

- Viewing test attachments on test result page:

Peek 2020-04-23 11-47

## Pre-Merge TODO
- [x] Add tests ([may be useful](https://github.com/jenkinsci/jenkins-test-harness));
- [x] Add some kind of CI;
- [x] Take a look at usecase when tests run on multiple slave nodes;
- [x] Remove `Screenshots and videos` when test doesn't have them;
- [x] Fix bug with multiple `TestData` (when user calls `junit` command more than once);
- [x] Split up `Screenshots and videos` into two different tables (`Screenshots` and `Videos`);
- [x] Show path to attachment relative to attachments dir;

### CODE REVIEW

**Comment by wentwrong** [comment]
After testing on multiple agents, I have realized that attachments with the same name from different agents override each other. So, new way of archiving looks like this:
1. Reporter write this to system-out of test:

```[[ATTACHMENT|/home/jenkins/agent/workspace/job/videos/2020-04-29_11-27-36/test-2/Firefox_75.0_Linux_0.0/1.mp4|cf3742e6-50e7-4117-9b3a-5160d4090292]]```

(format was expanded with UUID hash value for uniqueness when moved from agent to master)

2. junit parses produced report and calls our plugin.

3. Plugin moves attachment `/home/jenkins/agent/workspace/job/videos/2020-04-29_11-27-36/test-2/Firefox_75.0_Linux_0.0/1.mp4` to master with the name `cf3742e6-50e7-4117-9b3a-5160d4090292.mp4`.

pluginWork

Functionality described above was added in the [last commit](https://github.com/wentwrong/testcafe-jenkins-plugin/pull/1/commits/e78b690b7bff752462bb9eb2e1916f412a6cdee1)

**Comment by wentwrong** [comment]
## Examples
- No attachments
withoutAttachments
- One video attachment
videoAttachment
- One screenshot and video attachment
screensAndVideosAttachments

File: README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Jenkins Testcafe Plugin
+[![GitHub release](https://img.shields.io/github/v/release/wentwrong/testcafe-jenkins-plugin?include_prereleases)](https://github.com/wentwrong/testcafe-jenkins-plugin/releases)
+
+This plugin integrates <a href="https://devexpress.github.io/testcafe/">Testcafe</a> with Jenkins allowing to view attachments (screenshots and videos) directly on test page.
+
+## How-to
+This plugin should be used with [Testcafe Jenkins Reporter](https://github.com/wentwrong/testcafe-reporter-jenkins/).
+**Package is not yet published to npm registry, so use**
```
Comment (AndreyBelym) [inline README.md:8]: I guess the reporter is published now, isn't it?

File: src/main/java/io/jenkins/plugins/testcafe/TestcafeTestAction.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Override
+    public String getIconFileName() {
+        return "package.gif";
+    }
+
+    public String getUrl(Attachment attachment) {
+        return Jenkins.get().getRootUrl()
+                + testObject.getRun().getUrl()
+                + "testcafe-attachments/"
```
Comment (AndreyBelym) [inline src/main/java/io/jenkins/plugins/testcafe/TestcafeTestAction.java:59]: I think it's better to extract string literals to private static properties, especially used in expressions.

File: src/main/java/io/jenkins/plugins/testcafe/Attachment.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package io.jenkins.plugins.testcafe;
+
+public class Attachment {
+
+    private final String type;
+    private final String path;
+    private final String hashValue;
+
+    Attachment(String type, String path, String hashValue) {
+        this.type = type;
```
Comment (AndreyBelym) [inline src/main/java/io/jenkins/plugins/testcafe/Attachment.java:10]: Can we use an `enum` for the `type` property here? Or if a string is required due to serialization, can we extract possible values as a separate class?

**Comment by AndreyBelym** [comment]
@miherlosev, could you please review it too?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+This plugin integrates <a href="https://devexpress.github.io/testcafe/">Testcafe</a> with Jenkins allowing to view attachments (screenshots and videos) directly on test page.
+
+## How-to
+This plugin should be used with [Testcafe Jenkins Reporter](https://github.com/wentwrong/testcafe-reporter-jenkins/).
+```
+npm i testcafe-reporter-jenkins
+```
+### Usage on freestyle jobs
+Choose "Publish JUnit test result report" at Post-build Actions and "Testcafe Report" as additional test report features
```
[the remark this answers is not in the document]
Comment (wentwrong) [reply README.md]: localhost_8080_jenkins_job_configure

**Pull request merged**